Java/Gun/Gun.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Created by andrew on 03.12.16.
+ */
+public class Gun {
+    private int amount;
+    private int capacity;
+    private boolean isReady;
+    private String model;
+    private int totalShots;
+
+    public Gun(String model, int capacity) {
+        this.amount = 0;
+        this.capacity = capacity;
+        this.isReady = false;
+        this.model = model;
+        this.totalShots = 0;
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public int getCapacity() {
+        return capacity;
+    }
+
+    public String getModel() {
+        return model;
+    }
+
+    public int getTotalShots() {
+        return totalShots;
+    }
+
+    public boolean ready() {
+        return !isReady;
+    }
+
+    public void reload() {
+        amount = capacity;
+    }
+
+    public void prepare() {
+        isReady = !isReady;
+    }
+
+    public void shoot() throws OutOfRoundsException {
+        if ( amount == 0 ) {
+            throw new OutOfRoundsException();
+        }
+
+        amount -= 1;
+        totalShots += 1;
+        System.out.println("Bang!");
+        System.out.println("Total shots: ( " + getTotalShots() + "/" + getCapacity() + " )");
```

Comment:
Can also use `String.format("Total shots: (%d/%d)", getTotalShots(), getCapacity())`. It works as `printf()` in C. Sometimes it makes code more readable. Especially when you have a long String with large amount of arguments.